WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridFragment.java

```diff
@@ -142,9 +143,11 @@ public void onCreate(Bundle savedInstanceState) {
         if (savedInstanceState == null) {
             mSite = (SiteModel) getActivity().getIntent().getSerializableExtra(WordPress.SITE);
             mBrowserType = (MediaBrowserType) getActivity().getIntent().getSerializableExtra(MediaBrowserActivity.ARG_BROWSER_TYPE);
+            mImagesOnly = getActivity().getIntent().getBooleanExtra(MediaBrowserActivity.ARG_IMAGES_ONLY, false);
```

Comment:
Just a question, instead of introducing a new variable `mImagesOnly` here, can't we just set `mFilter = Filter.IMAGES;` when `getActivity().getIntent().getBooleanExtra(MediaBrowserActivity.ARG_IMAGES_ONLY, false);` is true?

And keep saveState and restoreState  without any change?